coderd/roles.go

```diff
@@ -13,31 +13,44 @@ import (
 
 // assignableSiteRoles returns all site wide roles that can be assigned.
 func (api *API) assignableSiteRoles(rw http.ResponseWriter, r *http.Request) {
-	// TODO: @emyrk in the future, allow granular subsets of roles to be returned based on the
-	// 	role of the user.
-
+	actorRoles := httpmw.AuthorizationUserRoles(r)
```

Comment:
Any specific reason we're doing this before `api.Authorize`?

Reply:
It is just pulling the roles from the actor from the middleware. This doesn't execute any code. Just pulling state from the request.